ui/src/tempVars/components/KeysTemplateBuilder.tsx

```diff
@@ -0,0 +1,241 @@
+  public async componentDidMount() {
+    await this.loadDatabases()
+    await this.loadMeasurements()
+    await this.loadKeys()
+  }
+
+  public render() {
+    const {queryPrefix} = this.props
+    const {
+      databases,
+      databasesStatus,
+      selectedDatabase,
+      measurements,
+      measurementsStatus,
+      selectedMeasurement,
+      keys,
+      keysStatus,
+    } = this.state
+
+    return (
+      <div className="temp-builder measurements-temp-builder">
+        <div className="form-group">
+          <label>Meta Query</label>
+          <div className="temp-builder--mq-controls">
+            <div className="temp-builder--mq-text">{queryPrefix}</div>
+            <DropdownPlaceholder rds={databasesStatus}>
+              <Dropdown
+                items={databases.map(text => ({text}))}
+                onChoose={this.handleChooseDatabaseDropdown}
+                selected={selectedDatabase}
+                buttonSize=""
+              />
+            </DropdownPlaceholder>
+            <div className="temp-builder--mq-text">FROM</div>
+            <DropdownPlaceholder rds={measurementsStatus}>
+              <Dropdown
+                items={measurements.map(text => ({text}))}
+                onChoose={this.handleChooseMeasurementDropdown}
+                selected={selectedMeasurement}
+                buttonSize=""
+              />
+            </DropdownPlaceholder>
+          </div>
+        </div>
+        <TemplateMetaQueryPreview items={keys} loadingState={keysStatus} />
+      </div>
+    )
+  }
+
+  private async loadDatabases() {
+    const {source} = this.props
+
+    this.setState({databasesStatus: RemoteDataState.Loading})
+
+    try {
+      const {data} = await showDatabases(source.links.proxy)
+      const {databases} = parseShowDatabases(data)
+      const {selectedDatabase} = this.state
+
+      this.setState({
+        databases,
+        databasesStatus: RemoteDataState.Done,
+      })
+
+      if (!selectedDatabase) {
+        this.handleChooseDatabase(_.get(databases, 0, ''))
+      }
+    } catch (error) {
+      this.setState({databasesStatus: RemoteDataState.Error})
+      console.error(error)
+    }
+  }
+
+  private async loadMeasurements() {
+    const {source} = this.props
+    const {selectedDatabase, selectedMeasurement} = this.state
+
+    this.setState({measurementsStatus: RemoteDataState.Loading})
+
+    try {
+      const {data} = await showMeasurements(
+        source.links.proxy,
+        selectedDatabase
+      )
+      const {measurementSets} = parseShowMeasurements(data)
+      const measurements = _.get(measurementSets, '0.measurements', [])
+
+      this.setState({
+        measurements,
+        measurementsStatus: RemoteDataState.Done,
+      })
+
+      if (!selectedMeasurement) {
+        this.handleChooseMeasurement(_.get(measurements, 0, ''))
+      }
+    } catch (error) {
+      this.setState({measurementsStatus: RemoteDataState.Error})
+      console.error(error)
```

Comment:
same question here about a possible user notification. not implying that there ought to be, just asking.